syncopy/tests/test_continuousdata.py

```diff
@@ -778,6 +785,157 @@ def test_sd_dataselection(self):
+            with pytest.raises(SPYValueError) as spyval:
+                operation(dummy, dummy2)
+                assert "Syncopy object with same number of trials (selected)" in str (spyval.value)
+            dummy2._selection = None
+
+            # Scalar algebra must be commutative (except for pow)
+            for operand in scalarOperands:
+                result = operation(dummy, operand) # perform operation from right
+                for tk, trl in enumerate(result.trials):
+                    assert np.array_equal(trl, operation(dummy.trials[tk], operand))
+                # Don't try to compute `2 ** data``
+                if operation(2,3) != 8:
+                    result2 = operation(operand, dummy) # perform operation from left
+                    assert np.array_equal(result2.data, result.data)
+
+                # Same as above, but swapped `dimord`
+                result = operation(ymmud, operand)
+                for tk, trl in enumerate(result.trials):
+                    assert np.array_equal(trl, operation(ymmud.trials[tk], operand))
+                if operation(2,3) != 8:
+                    result2 = operation(operand, ymmud)
+                    assert np.array_equal(result2.data, result.data)
+
+            # Careful: NumPy tries to avoid failure by broadcasting; instead of relying
+            # on an existing `__radd__` method, it performs arithmetic component-wise, i.e.,
+            # ``np.ones((3,3)) + data`` performs ``1 + data`` nine times, so don't
+            # test for left/right arithmetics...
+            for operand in dummyOperands:
+                result = operation(dummy, operand)
+                for tk, trl in enumerate(result.trials):
+                    assert np.array_equal(trl, operation(dummy.trials[tk], operand))
+            for operand in ymmudOperands:
+                result = operation(ymmud, operand)
+                for tk, trl in enumerate(result.trials):
+                    assert np.array_equal(trl, operation(ymmud.trials[tk], operand))
+
+            # Ensure erroneous object type-casting is prevented
+            with pytest.raises(SPYTypeError) as spytyp:
+                operation(dummy, dummyC)
+                assert "Syncopy data object of same numerical type (real/complex)" in str(spytyp.value)
+
+            # Most severe safety hazard: throw two objects at each other (with regular and
+            # swapped dimord)
+            result = operation(dummy, dummy2)
+            for tk, trl in enumerate(result.trials):
+                assert np.array_equal(trl, operation(dummy.trials[tk], dummy2.trials[tk]))
+            result = operation(ymmud, ymmud2)
+            for tk, trl in enumerate(result.trials):
+                assert np.array_equal(trl, operation(ymmud.trials[tk], ymmud2.trials[tk]))
+
+            # Now the most complicated case: user-defined subset selections are present
+            for trialSel in trialSelections:
```

Comment:
This looks a bit like over-engineering to me.. especially the taper part

Reply:
Yupp, this really tests everything and the kitchen sink but at the expense of significant test runtime. We might want to consider simplifying the selection test cases here. Thinking about a clever but still exhaustive test setup...